hols/todo/MultiCluster.md

```diff
@@ -0,0 +1,357 @@
+6. Check 'Enable cross-region backup replication to disaster recovery peer'
+7. Click on 'Add Peer Database' to finish creating the Autonomous DR instance
+
+## C. Kubernetes Operators
+
+We will use 2 Kubernetes Operators:
+
+1. Oracle Coherence Operator: this will be used to deploy the Coherence cluster to OKE.
+2. TODO: External Secrets Operator: this will be used to retrieve the Hibernate Configuration and Wallet Secret from OCI Vault.
+
+### Deploy Coherence Operator
+
+1. Add Helm chart repository for Coherence Operator:
+```bash
+helm repo add coherence https://oracle.github.io/coherence-operator/charts
+helm repo update
+```
+
+2. Install Coherence Operator:
+```bash
+for c in c1 c2; do
+  kubectx $c
+  helm install --namespace coherence-operator coherence coherence/coherence-operator --create-namespace  
+done
+```
+### Prepare Kubernetes Namespaces, ConfigMaps and Secrets
+
+1. Create and label a namespace for the todo application in Kubernetes:
+
+```bash
+for c in c1 c2; do
+  kubectx $c
+  kubectl create ns todo
+  kubectl label namespace todo istio-injection=enabled
+done
+``` 
+
+2. Create a secret to authenticate with OCIR:
+
+```bash
+for c in c1 c2; do
+  kubectx $c
+  kubectl create secret -n todo docker-registry ocir-secret --docker-server=<your-registry-server> --docker-username=<your-name> --docker-password=<your-pword> --docker-email=<your-email>
+done
+```
+
+3. Download the Database Wallet and upload it to the database folder on the operator host:
+
+```bash
+scp Wallet_TasksDB.zip operator:~/database/
+```
+
+[//]: # (4. Edit the tnsnames.ora and replace `<servicename>` and the region identifiers for your database as appropriate:)
+
+[//]: # ()
+[//]: # (```txt)
+
+[//]: # (tasksdb_medium = &#40;description=)
+
+[//]: # (                        &#40;failover=on&#41;)
+
+[//]: # (                        &#40;retry_count=20&#41;&#40;retry_delay=3&#41;&#40;connect_timeout=120&#41;&#40;transport_connect_timeout=3&#41;)
+
+[//]: # (                        &#40;address_list=&#40;load_balance=on&#41;)
+
+[//]: # (                                &#40;address=&#40;protocol=tcps&#41;&#40;port=1522&#41;&#40;host=adb.ap-sydney-1.oraclecloud.com&#41;&#41;&#41;)
+
+[//]: # (                        &#40;address_list=&#40;load_balance=on&#41;)
+
+[//]: # (                                &#40;address=&#40;protocol=tcps&#41;&#40;port=1522&#41;&#40;host=adb.ap-melbourne-1.oraclecloud.com&#41;&#41;&#41;)
+
+[//]: # (                        &#40;connect_data=&#40;service_name=<servicename>_tasksdb_medium.adb.oraclecloud.com&#41;&#41;)
+
+[//]: # (                        &#40;security=&#40;ssl_server_dn_match=yes&#41;&#41;&#41;&#41;)
+
+[//]: # (```)
+
+4. Extract the Wallet and switch the tnsnames files:
+
+```bash
+cd database
+unzip Wallet_TasksDB.zip
+mv tnsnames.ora tnsnames.ora.orig
+mv tnsnames.ora.ha tnsnames.ora
+```
+ 
+5. Create a secret to store the wallet:
+
+```bash
+export OJDBC=$HOME/database/ojdbc.properties
```

Comment:
Instead of $HOME, we should assume the current working directory.
E.g.
```shell
unzip Wallet_TasksDB.zip -d wallet
export OJDBC=$PWD/wallet/ojdbc.properties
```

Reply:
nope, sometimes a step is missed and a user comes back to it and the directory is not the same